Add OpenID Connect authentication option

MWY3510 opened the pull request:
This pull request adds OpenID Connect auth using the python-social-auth project. As a result it also bumps the minor minor version of Django.
Environment variables are added to control OpenID auth and to show or not show the password auth form for users.
This should resolve issue: #386 
I can add documentation later this week in a follow up request if you wish.
Let me know if you would like any changes or want a different approach to this issue.

Sispheor commented on Squest/settings.py:
```diff
@@ -30,6 +30,12 @@
 LDAP_ENABLED = str_to_bool(os.environ.get('LDAP_ENABLED', False))
+OPENID_ENABLED = str_to_bool(os.environ.get('OPENID_ENABLED', False))
+OPENID_BTN_TEXT = os.environ.get('OPENID_BTN_TEXT', 'OpenID Login')
+SOCIAL_AUTH_OIDC_OIDC_ENDPOINT = os.environ.get('OIDC_ENDPOINT', '')
```
We should keep python variable name and env var align to be consistent.

Also all of this should be placed in the doc.

I would go with the same prefix for all variable that concern OIDC.

MWY3510 replied:
I can do that. Should have that shortly.

Sispheor commented on templates/registration/login.html:
```diff
@@ -39,6 +40,14 @@
+                    {% if openid_active %}
+                    <div class="card-footer">
+                        <div class="text-center">
+                            <a class="btn" style="background-color: #6ad3c2" href="{% url "social:begin" "oidc" %}" role="button"><strong>{{ openid_btn_text }}</strong></a>
```
Could we use a pre [defined color](https://adminlte.io/docs/3.0/layout.html) of adminlte framework?

MWY3510 replied:
Just to check, since this would make it inconsistent with the password form sign-in button. Do you want that one adjusted as well?

Sispheor replied:
Oh sorry I haven't seen  that it was like that for the first buton.
You can leave it like this I guess then.

Sispheor commented:
Hi, thanks for the PR. I left couple notes.
Could you also squash your commits into a single one?
Thanks.

MWY3510 commented:
Adjusted the variable names and added some documentation. I am waiting on the commit squash until I hear back on the style review question. Thank you.

Sispheor commented:
All good. Except that we've released a new version and so this need to be rebased again on the last dev.
I recommend you to access was is one the remote concerning poetry and then run the command line to install back your libs so the poetry lock will be ok.

EliasBoulharts reviewed (approved):
LGTM, sorry I forgot to change my upstream and I had force push in your branch. I rebased the branch, I squashed the commits and changed the variable that was missing from the settings. I tested it and it works well, good job!

a-belhadj commented on Squest/settings.py:
```diff
@@ -124,6 +136,25 @@
     },
 ]
 
+if SOCIAL_AUTH_OIDC_ENABLED and not TESTING:
+    TEMPLATES = [
+        {
+            'BACKEND': 'django.template.backends.django.DjangoTemplates',
+            'DIRS': [TEMPLATES_DIR],
+            'APP_DIRS': True,
+            'OPTIONS': {
+                'context_processors': [
+                    'django.template.context_processors.debug',
+                    'django.template.context_processors.request',
+                    'django.contrib.auth.context_processors.auth',
+                    'social_django.context_processors.backends',
+                    'social_django.context_processors.login_redirect',
+                    'django.contrib.messages.context_processors.messages',
+                ],
+            },
+        },
+    ]
```
Use this instead, it will avoid code duplication :
```py
TEMPLATES['OPTIONS']['context_processors'].append('social_django.context_processors.backends')
TEMPLATES['OPTIONS']['context_processors'].append('social_django.context_processors.login_redirect')
```

MWY3510 replied:
This should be fixed now.

MWY3510 commented:
Added changes requested by @a-belhadj and squashed commits. @EliasBoulharts not a problem, thank you for reviewing. I plan on having additional pull requests soon anyway so I am not too worried.

EliasBoulharts commented:
@MWY3510 thanks again for your contribution, it will be very useful for us and the community! 
Hope to see you soon for new PRs 😉
